tests/scripts/generate_test_code.py

```diff
@@ -0,0 +1,752 @@
+
+    :param deps: List of dependencies.
+    :return: ifdef code
+    """
+    defines = ('#if ' if len(deps) else '') + ' && '.join(['%sdefined(%s)' % split_dep(x) for x in deps])
+    return defines
+
+
+def gen_function_wrapper(name, locals, args_dispatch):
+    """
+    Creates test function wrapper code. A wrapper has the code to unpack parameters from parameters[] array.
+
+    :param name: Test function name
+    :param locals: Local variables declaration code
+    :param args_dispatch: List of dispatch arguments. Ex: ['(char *)params[0]', '*((int *)params[1])']
+    :return: Test function wrapper.
+    """
+    # Then create the wrapper
+    wrapper = '''
+void {name}_wrapper( void ** params )
+{{
+{unused_params}{locals}
+    {name}( {args} );
+}}
+'''.format(name=name,
+           unused_params='' if args_dispatch else '    (void)params;\n',
+           args=', '.join(args_dispatch),
+           locals=locals)
+    return wrapper
+
+
+def gen_dispatch(name, deps):
+    """
+    Generates dispatch code for the test function table.
+
+    :param name: Test function name
+    :param deps: List of dependencies
+    :return: Dispatch code.
+    """
+    if len(deps):
+        ifdef = gen_deps_one_line(deps)
+        dispatch_code = '''
+{ifdef}
+    {name}_wrapper,
+#else
+    NULL,
+#endif
+'''.format(ifdef=ifdef, name=name)
+    else:
+        dispatch_code = '''
+    {name}_wrapper,
+'''.format(name=name)
+
+    return dispatch_code
+
+
+def parse_until_pattern(funcs_f, end_regex):
+    """
+    Parses function headers or helper code until end pattern.
+
+    :param funcs_f: file object for .functions file
+    :param end_regex: Pattern to stop parsing
+    :return: Test suite headers code
+    """
+    headers = '#line %d "%s"\n' % (funcs_f.line_no + 1, funcs_f.name)
+    for line in funcs_f:
+        if re.search(end_regex, line):
+            break
+        headers += line
+    else:
+        raise InvalidFileFormat("file: %s - end pattern [%s] not found!" % (funcs_f.name, end_regex))
+
+    return headers
+
+
+def parse_suite_deps(funcs_f):
+    """
+    Parses test suite dependencies.
+
+    :param funcs_f: file object for .functions file
+    :return: List of test suite dependencies.
+    """
+    deps = []
+    for line in funcs_f:
+        m = re.search('depends_on\:(.*)', line.strip())
+        if m:
+            deps += [x.strip() for x in m.group(1).split(':')]
+        if re.search(END_DEP_REGEX, line):
+            break
+    else:
+        raise InvalidFileFormat("file: %s - end dependency pattern [%s] not found!" % (funcs_f.name, END_DEP_REGEX))
+
+    return deps
+
+
+def parse_function_deps(line):
+    """
+    Parses function dependencies.
+
+    :param line: Line from .functions file that has dependencies.
+    :return: List of dependencies.
+    """
+    deps = []
+    m = re.search(BEGIN_CASE_REGEX, line)
+    dep_str = m.group(1)
+    if len(dep_str):
+        m = re.search('depends_on:(.*)', dep_str)
```

Comment:
I like your suggestions. A stronger regex will also help debug the .function file since `#line` directive is used after generated dependency code and will not indicate correct place in the generated code in case of error.